app/controllers/api/api_responder.rb

```diff
@@ -5,8 +5,20 @@ def api_behavior(error)
       raise error unless resourceful?
 
       if !get? && !post?
+        api_version = "v2"
+        api_version = "v1" if @controller.class.name.include?("V1")
         #return resource instead of default "head :no_content" for PUT, PATCH, and DELETE
-        display resource
+        #v1 puppetclasses do not have a working rabl template for "show" or "update"
+        unless delete? || (resource.class.name.include?("Puppetclass") && api_version == "v1")
+          path = resource.class.name
+          path = "hosts" if path.include?("Host::Managed") || path.include?("Host::Base")
```

Comment:
This method isn't really suitable, sorry.  Can you look at implementing it a different way?

In particular, consider behaviour for a plugin which wouldn't be able to modify this method, we need a more generic way if possible - something either in the controllers or views themselves probably.